### ISSUE TITLE
Malformed Request Body For Large Post Bodies

### ISSUE THREAD

**Issue by kira98k** [eclipse-ee4j/jersey#4796]
The below issue occurs in the latest version '3.0.2'. These are my dependencies :
```gradle
dependencies {
    compile "org.glassfish.jersey.core:jersey-client:3.0.2"
    implementation 'org.glassfish.jersey.inject:jersey-hk2:3.0.2'
    testImplementation 'org.junit.jupiter:junit-jupiter-api:5.6.0'
    testRuntimeOnly 'org.junit.jupiter:junit-jupiter-engine'
}
```
When the below post request is executed the request body received at the server is malformed for any entity with length greater than or equal to 65840.

```java
import jakarta.ws.rs.client.Client;
import jakarta.ws.rs.client.ClientBuilder;
import jakarta.ws.rs.client.Entity;
import jakarta.ws.rs.core.Response;

public class MyClass {
    public static void main(String[] args) {
        Client client = ClientBuilder.newClient();
//        int count = 65839; // Works
        int count = 65840; // Doesn't work
        String body = new String(new char[count]).replace("\0", "A");
        assert (body.length() == count);
        Response response = null;
        try {
            response = client.target("http://localhost:8888/")
                    .request()
                    .post(Entity.text(body));
        } catch (Exception e) {
            e.printStackTrace();
        }
    }
}
```
To reproduce the issue run a server, `nc -l 8888 > output.txt` and then run the java code.
The output looks like the following:
```http
AAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAPOST / HTTP/1.1
Content-Type: text/plain
User-Agent: Jersey/3.0.2 (HttpUrlConnection 1.8.0_212)
Host: localhost:8888
Accept: text/html, image/gif, image/jpeg, *; q=.2, */*; q=.2
Connection: keep-alive
Content-Length: 65840

AAAAAAAAAAAAAAAAA... (Rest Of The Body)
```
While for any body with length less than 65840, the output looks like:

```http
POST / HTTP/1.1
Content-Type: text/plain
User-Agent: Jersey/3.0.2 (HttpUrlConnection 1.8.0_212)
Host: localhost:8888
Accept: text/html, image/gif, image/jpeg, *; q=.2, */*; q=.2
Connection: keep-alive
Content-Length: 65839

AAAAAAAAAAAAAAAAA... (Rest Of The Body)
```

**Comment by jbescos** [comment]
I am testing this in Jersey (having server and client in both sides) and it is working fine. Maybe the problem is in 'nc' application. It is possible it has a limit to read new lines.

```
public class Issue4796Test extends JerseyTest {

    @Path("example")
    public static class Issue4796Resource {
        @POST
        public int post(String content) {
            return content.length();
        }
    }

    @Override
    protected Application configure() {
        return new ResourceConfig(Issue4796Resource.class);
    }

    @Test
    public void bigInputDoesntWork() {
        int count = 65840;
        String body = new String(new char[count]).replace("\0", "A");
        assert (body.length() == count);
        Response response = target("/example").request().post(Entity.text(body));
        int value = response.readEntity(Integer.class);
        assertEquals(count, value);
    }

    @Test
    public void bigInputWork() {
        int count = 65839;
        String body = new String(new char[count]).replace("\0", "A");
        assert (body.length() == count);
        Response response = target("/example").request().post(Entity.text(body));
        int value = response.readEntity(Integer.class);
        assertEquals(count, value);
    }
}
```

**Comment by kardeepak** [comment]
Hi, 
I tried the above test on my local as well as in a linux container. It fails on my macbook but it runs successfully in the linux container. This is very weird.
Can any verify it on a mac ?

**Comment by kira98k** [comment]
Hi,
I tried the above test on my local as well as in a linux container. It fails on my macbook but it runs successfully in the linux container. This is very weird.
Can any verify it on a mac ?

**Comment by senivam** [comment]
@kira98k, I'll take a look

**Comment by senivam** [comment]
for now I can only state that it works basically everywhere. I've used your example and ``nc -l 8888 > output.txt`` as a reproducer and everywhere it passes OK. 

I've checked both on 3.x (3.1.0-SNPASHOT) and 3.0.2 versions and everywhere it passes OK.

My command line as follows:

```
java -cp .:lib/jersey-client-3.0.2.jar:lib/jakarta.ws.rs-api-3.0.0.jar:lib/jersey-common-3.0.2.jar:lib/jersey-hk2-3.0.2.jar:lib/jakarta.annotation-api-2.0.0.jar:lib/hk2-api-3.0.1.jar:lib/hk2-utils-3.0.1.jar:lib/hk2-locator-3.0.1.jar:lib/jakarta.inject-api-2.0.0.jar:lib/jakarta.activation-2.0.0.jar org.glassfish.jersey.examples.nc.MyClass
```
the whole CP is listed to show that only required libraries were included. 

class ``org.glassfish.jersey.examples.nc.MyClass`` is copy of your reproducer. 

_nc_ was running on port 8888 meanwhile. I was using _nc_ and _ncat_ to check if an app is to blame. But for both it works. 

For now there are no signs that something is wrong with Jersey.
